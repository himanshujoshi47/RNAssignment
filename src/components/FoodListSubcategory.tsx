import React from 'react';
import {View, StyleSheet, Text} from 'react-native';
import {Subcategory} from '../network/models/FoodListModel';
import R from '../res';
import FoodListSubcategoryTitleCell from './FoodListSubcategoryTitleCell';

interface FoodListSubcategoryProps {
  quote: string;
  colorCode: string;
  subcategories: Subcategory[];
}

const FoodListSubcategory: React.FC<FoodListSubcategoryProps> = ({
  quote,
  colorCode,
  subcategories,
}) => {
  return (
    <View
      style={{
        paddingHorizontal: R.dimensions.dimen_4,
        paddingTop: R.dimensions.dimen_4,
      }}>
      {subcategories &&
        subcategories.map((item, index) => {
          return (
            <View key={`subcat-title-${index}`}>
              {item.subCategoryname.length > 0 && (
                <Text
                  testID={R.id.FOOD_LIST_SUBCAT_TITLE}
                  style={[localStyles.subCatName, {color: colorCode}]}>
                  {item.subCategoryname.toUpperCase()}
                </Text>
              )}
              <View>
                {item.items &&
                  item.items.map((subCat, subCatIndex) => {
                    return (
                      <FoodListSubcategoryTitleCell
                        key={`subcat-title-${index}-subcat-name-${subCatIndex}`}
                        title={subCat}
                      />
                    );
                  })}
                {quote.length > 0 && (
                  <View style={localStyles.quoteViewContainer}>
                    <Text style={localStyles.quoteText}>{quote}</Text>
                  </View>
                )}
              </View>
            </View>
          );
        })}
    </View>
  );
};

const localStyles = StyleSheet.create({
  subCatName: {
    fontWeight: 'bold',
    fontSize: R.dimensions.dimen_18,
    marginTop: R.dimensions.dimen_24,
    marginBottom: R.dimensions.dimen_15,
    color: R.colors.text.primary,
  },
  quoteViewContainer: {
    marginVertical: R.dimensions.dimen_8,
    padding: R.dimensions.dimen_8,
  },
  quoteText: {
    paddingVertical: R.dimensions.dimen_20,
    paddingHorizontal: R.dimensions.dimen_8,
    textAlign: 'center',
    fontSize: R.dimensions.dimen_12,
    backgroundColor: R.colors.background.green,
    borderRadius: R.dimensions.dimen_8,
    overflow: 'hidden',
    fontWeight: '300',
    color: R.colors.text.primary,
  },
});

export default FoodListSubcategory;
