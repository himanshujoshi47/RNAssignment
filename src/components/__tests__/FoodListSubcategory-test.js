import React from 'react';
import renderer from 'react-test-renderer';
import R from '../../res';
import FoodListSubcategory from '../FoodListSubcategory';

jest.useFakeTimers();

describe('renders and have required elements', () => {
  let props = {
    colorCode: '#333',
    subcategories: [
      {
        items: [
          'Beef (93/7 ground)',
          'Bison',
          'Filet Mignon (trim fat)',
          'Kangaroo',
          'Sirloin (ground)',
          'Sirloin Strip Steak',
          'Veal (Ground)',
          'Venison',
        ],
        subCategoryname: 'Beef',
      },
    ],
  };

  const foodListSubcategory = renderer.create(
    <FoodListSubcategory {...props} />,
  );
  const foodListSubcategoryInst = foodListSubcategory.root;

  it('renders correctly', () => {
    expect(foodListSubcategory).toMatchSnapshot();
  });

  it('should have required elements', () => {
    expect(
      foodListSubcategoryInst.findByProps({
        testID: R.id.FOOD_LIST_SUBCAT_TITLE,
      }),
    ).toBeTruthy();
  });
});
