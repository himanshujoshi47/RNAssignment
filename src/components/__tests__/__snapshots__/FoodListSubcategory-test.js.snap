// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and have required elements renders correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 4,
      "paddingTop": 4,
    }
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#222222",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 15,
            "marginTop": 24,
          },
          Object {
            "color": "#333",
          },
        ]
      }
      testID="food-list-subcat-title"
    >
      BEEF
    </Text>
    <View>
      <View>
        <Text
          style={
            Object {
              "color": "#222222",
              "fontSize": 18,
              "marginBottom": 16,
              "marginTop": 15,
            }
          }
          testID="food-list-item-title"
        >
          Beef (93/7 ground)
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ECECEC",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#222222",
              "fontSize": 18,
              "marginBottom": 16,
              "marginTop": 15,
            }
          }
          testID="food-list-item-title"
        >
          Bison
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ECECEC",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#222222",
              "fontSize": 18,
              "marginBottom": 16,
              "marginTop": 15,
            }
          }
          testID="food-list-item-title"
        >
          Filet Mignon (trim fat)
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ECECEC",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#222222",
              "fontSize": 18,
              "marginBottom": 16,
              "marginTop": 15,
            }
          }
          testID="food-list-item-title"
        >
          Kangaroo
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ECECEC",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#222222",
              "fontSize": 18,
              "marginBottom": 16,
              "marginTop": 15,
            }
          }
          testID="food-list-item-title"
        >
          Sirloin (ground)
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ECECEC",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#222222",
              "fontSize": 18,
              "marginBottom": 16,
              "marginTop": 15,
            }
          }
          testID="food-list-item-title"
        >
          Sirloin Strip Steak
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ECECEC",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#222222",
              "fontSize": 18,
              "marginBottom": 16,
              "marginTop": 15,
            }
          }
          testID="food-list-item-title"
        >
          Veal (Ground)
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ECECEC",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#222222",
              "fontSize": 18,
              "marginBottom": 16,
              "marginTop": 15,
            }
          }
          testID="food-list-item-title"
        >
          Venison
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ECECEC",
              "height": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
